src/mx/TextFileSource.java

```diff
@@ -0,0 +1,45 @@
+package mx;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class TextFileSource implements SourceInterface {
+
+    protected ArrayList<Domain> domains;
+
+    protected String filename;
+
+    public TextFileSource(String inputFilename) {
+        domains = new ArrayList<>();
+        filename = inputFilename;
+    }
+
+    @Override
+    /**
+     * @throws SourceNotFoundException
+     */
+    public ArrayList<Domain> getDomainList() throws SourceNotFoundException {
+        domains.clear();
```

Comment:
What will be use of that class - will it work as a watcher, or should it just read a file? Most probably you don't want read the file on every invokation of getter. It's actually confusing, because user of this method won't know that it reads file, unless he reads the source. Would be much better to cache this state and return it ever since